bindings/matrix-sdk-crypto-js/tests/machine.test.js

```diff
@@ -622,4 +627,182 @@ describe(OlmMachine.name, () => {
+                            ],
+                            "keys": {
+                                "ed25519:GLhEKLQ50jnF6IMEPsO2ucpHUNIUEnbBXs5gYbHg4Aw": "GLhEKLQ50jnF6IMEPsO2ucpHUNIUEnbBXs5gYbHg4Aw"
+                            },
+                            "signatures": {
+                                "@example:morpheus.localhost": {
+                                    "ed25519:ZzU4WCyBfOFitdGmfKCq6F39iQCDk/zhNNTsi+tWH7A": "4fIyWlVzuz1pgoegNLZASycORXqKycVS0dNq5vmmwsVEudp1yrPhndnaIJ3fjF8LDHvwzXTvohOid7DiU1j0AA"
+                                }
+                            }
+                        }
+                    }
+                });
+                const marked = await m.markRequestAsSent('foo', RequestType.KeysQuery, hypothetical_response);
+            }
+        });
+
+        test('can start an in-room SAS verification', async () => {
+            let _ = m.bootstrapCrossSigning(true);
+            const identity = await m.getIdentity(new UserId('@example:morpheus.localhost'));
+
+            expect(identity).toBeInstanceOf(UserIdentity);
+            expect(identity.isVerified()).toStrictEqual(false);
+
+            const eventId = new EventId('$Rqnc-F-dvnEYJTyHq_iKxU2bZ1CI92-kuZq3a5lr5Zg');
+            const verificationRequest = await identity.requestVerification(room, eventId);
+            expect(verificationRequest).toBeInstanceOf(VerificationRequest);
+
+            await m.receiveVerificationEvent(
+                JSON.stringify({
+                    "sender": "@example:morpheus.localhost",
+                    "type": "m.key.verification.ready",
+                    "event_id": "$QguWmaeMt6Hao7Ea6XHDInvr8ndknev79t9a2eBxlz0",
+                    "origin_server_ts": 1674037263075,
+                    "content": {
+                        "methods": [
+                            "m.sas.v1",
+                            "m.qr_code.show.v1",
+                            "m.reciprocate.v1"
+                        ],
+                        "from_device": "SUMODVLSIU",
+                        "m.relates_to": {
+                            "rel_type": "m.reference",
+                            "event_id": eventId.toString(),
+                        }
+                    }
+                }),
+                room
+            );
+
+            expect(verificationRequest.roomId.toString()).toStrictEqual(room.toString());
+
+            const [_sas, outgoingVerificationRequest] = await verificationRequest.startSas();
+
+            expect(outgoingVerificationRequest).toBeInstanceOf(RoomMessageRequest);
+            expect(outgoingVerificationRequest.id).toBeDefined();
+            expect(outgoingVerificationRequest.room_id).toStrictEqual(room.toString());
+            expect(outgoingVerificationRequest.txn_id).toBeDefined();
+            expect(outgoingVerificationRequest.event_type).toStrictEqual('m.key.verification.start');
+            expect(outgoingVerificationRequest.content).toBeDefined();
```

Comment:
please can we test what's inside `content`? It should have properties like `from_device` (see https://spec.matrix.org/v1.5/client-server-api/#mkeyverificationstart).

Reply:
Sure!